src/InlineStrings.jl

```diff
@@ -616,6 +619,38 @@ Base.string(a::BaseStrs, b::BaseStrs, c::InlineString) = _string(a, b, c)
     return out
 end
 
+# For more and/or bigger InlineStrings creating a `Base.String` is faster
+const TinyInlineStrings = Union{InlineString1, InlineString3, InlineString7}
+Base.string(a::TinyInlineStrings, b::TinyInlineStrings, c::TinyInlineStrings) = _string(_string(a, b), c)
+
+# Only benefit from keeping the small-ish strings as InlineStrings
+function _string(a::Ta, b::Tb) where {Ta <: SmallInlineStrings, Tb <: SmallInlineStrings}
+    T = summed_type(Ta, Tb)
+    sz_a = Int(!isa(a, InlineString1))
+    sz_b = Int(!isa(b, InlineString1))
+    len_a = sizeof(a)
+    len_b = sizeof(b)
+    a2 = Base.shl_int(Base.zext_int(T, Base.lshr_int(a, 8*sz_a)), 8 * (sizeof(T) - sizeof(Ta) + sz_a))
+    b2 = Base.shl_int(Base.zext_int(T, Base.lshr_int(b, 8*sz_b)), 8 * (sizeof(T) - sizeof(Tb) + sz_b - len_a))
+    len = _oftype(T, len_a + len_b)
+    # @show bitstring(a2)
+    # @show bitstring(b2)
+    # @show bitstring(len)
+    return Base.or_int(Base.or_int(a2, b2), len)
+end
+
+summed_type(::Type{InlineString1}, ::Type{InlineString1}) = InlineString3
+summed_type(::Type{InlineString3}, ::Type{InlineString1}) = InlineString7
+summed_type(::Type{InlineString3}, ::Type{InlineString3}) = InlineString7
+summed_type(::Type{InlineString7}, ::Type{InlineString1}) = InlineString15
+summed_type(::Type{InlineString7}, ::Type{InlineString3}) = InlineString15
+summed_type(::Type{InlineString7}, ::Type{InlineString7}) = InlineString15
+summed_type(::Type{InlineString15}, ::Type{InlineString1}) = InlineString31
+summed_type(::Type{InlineString15}, ::Type{InlineString3}) = InlineString31
+summed_type(::Type{InlineString15}, ::Type{InlineString7}) = InlineString31
+summed_type(::Type{InlineString15}, ::Type{InlineString15}) = InlineString31
+summed_type(a::Type{<:InlineString}, b::Type{<:InlineString}) = summed_type(b, a)
```

Comment:
do we need this definition? it seems we're only dispatching on small inline string types so we won't hit here ever?

Reply:
we just need this to avoid writing out methods for each order - you're right, this can be limited to small types
```suggestion
summed_type(a::Type{<:SmallInlineString}, b::Type{<:SmallInlineString}) = summed_type(b, a)
```